pkg/cloud/services/gc/utils.go

```diff
@@ -0,0 +1,56 @@
+package gc
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+)
+
+const (
+	fakePartition     = "aws"
+	fakeRegion        = "fake-region"
+	fakeAccount       = "fake-account"
+	elbService        = "elasticloadbalancing"
+	elbResourcePrefix = "loadbalancer/"
+	sgService         = "ec2"
+	sgResourcePrefix  = "security-group/"
+
+	// maxDescribeTagsRequest is the maximum number of resources for the DescribeTags API call
+	// see: https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_DescribeTags.html.
+	maxDescribeTagsRequest = 20
+)
+
+// composeArn composes a resource arn with correct service and resource, but fake partition, region and account.
+func composeArn(service, resource string) string {
+	return "arn:" + fakePartition + ":" + service + ":" + fakeRegion + ":" + fakeAccount + ":" + resource
+}
+
+// composeAWSResource composes *AWSResource object for an aws resource.
+func composeAWSResource(resourceARN *string, resourceTags infrav1.Tags) (*AWSResource, error) {
+	parsedArn, err := arn.Parse(*resourceARN)
+	if err != nil {
+		return nil, fmt.Errorf("parsing resource arn %s: %w", *resourceARN, err)
+	}
+
+	resource := &AWSResource{
+		ARN:  &parsedArn,
+		Tags: resourceTags,
+	}
+
+	return resource, nil
+}
+
+// chunkResources is similar to https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/elb/loadbalancer.go#L1488.
```

Comment:
Thanks for leaving this comment. Could you also add a similar one to https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/elb/loadbalancer.go#L1488?

If someone needs to copy either one of these in the future, they'll know there will be 3 places where such a function is need, and will decide to factor it out into a separate package.

Reply:
OK. Added!